src/components/Search/index.tsx

```diff
@@ -0,0 +1,161 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import './style.css';
+import { useState } from 'react';
+
+//Hooks
+import { useGet } from '../../hooks/get';
+
+//Components
+import Counter from '../../components/Counter';
+
+//DatePicker
+import DatePicker from 'react-datepicker';
+import "react-datepicker/dist/react-datepicker.css";
+
+
+//Imag
+import departure from '../../assets/Flights/departure.png'
+import arrival from '../../assets/Flights/arrival.png'
+import personsolid from '../../assets/Flights/personsolid.png'
+import calendarwithdates from '../../assets/Flights/calendarwithdates.png'
+import Button from '../../components/Button';
+import { useSearchContext } from '../../contexts/searchContext';
+
+export default function Search() {
+
+
+ /*********************Interface****************************/
+
+  interface typeOfStateClick{
+    [arg:string]:boolean,
+  }
+
+  /*********************State****************************/
+  
+  const [stateClick, setStateClick] = useState<typeOfStateClick>({
+    'fromWhere':false,
+    'whereTo':false,
+    'date':false,
+    'passenger':false,
+  });
+  const [checkbox, setCheckbox] = useState<'roundTrip' | 'oneWay'>('roundTrip');
+  const {setSearch} = useSearchContext();
+  const [fromWhere, setFromWhere] = useState<string>('From where?'); 
+  const [whereTo, setWhereTo] = useState<string>('Where To?'); 
+  const [adult, setAdult] = useState<number | "1 adult">("1 adult"); 
+  const [minors, setMinor] = useState<number>(0);   
+  const [startDate, setStartDate] = useState(new Date()); 
+  const [endDate, setEndDate] = useState(new Date()); 
+  const [defaultDate] = useState(new Date()); 
```

Comment:
All of this is extremely confusing, from the interface which is declared inside the component to the states.
- The interface and generally any constants must not be declared inside functions, you don't want this code to be rerun every time the function runs or rerenders, you just want it to run once since it is constant and unchanging.
- Your whole approach and thinking in this component is indicative of code-first plan-never approach, this component is clearly a form, you need only one state, the form state, which includes all of these values and variables in a single object, because they are representing a single concept.